staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -825,6 +825,14 @@ func StrategicMergeMapPatch(original, patch JSONMap, dataStruct interface{}) (JS
 	if err != nil {
 		return nil, err
 	}
+
+	// Because custom resources are decoded as Unstructured and because we're missing the metadata about how to handle
+	// each field in a strategic merge patch, we can't easily do a strategic merge for custom resources.
+	// So we should fail fast and return an error.
+	if t.Name() == "Unstructured" {
+		return nil, fmt.Errorf("cannot apply strategic merge patch for custom resources, try --type merge")
```

Comment:
Isn't the strategy patch logic without any metadata just the classical non-strategic patch? What does the user see now?

In general, I would prefer not to mention custom resources here in apimachinery. An error constant in vendor/k8s.io/apimachinery/pkg/util/mergepatch/errors.go with "ErrUnsupportedPatchFormat" would make more sense.

Reply:
Done.